deployments/pulse3d/services/pulse3d/src/main.py

```diff
@@ -701,70 +702,29 @@ async def get_interactive_waveform_data(
-            time = time_force_df[columns[0]].tolist()
-
-            # set up empty dictionaries to be passed in response
-            coordinates = dict()
-            peaks_and_valleys = dict()
-            for well in columns:
-                if not _is_valid_well_name(well):
-                    continue
-
-                well_force = time_force_df[well].dropna().tolist()
-                if peaks_valleys_needed:
-                    if needs_unit_conversion:
-                        # not exact, but this isn't used outside of graphing in FE, real raw data doesn't get changed
-                        min_value = min(well_force)
-                        well_force -= min_value
-                        well_force *= MICRO_TO_BASE_CONVERSION
-                        time = [i * MICRO_TO_BASE_CONVERSION for i in time]
-
-                    interpolated_well_data = np.row_stack([time[: len(well_force)], well_force])
-
-                    peak_detector_params = {
-                        param: analysis_params[param]
-                        for param in ("prominence_factors", "width_factors", "start_time", "end_time")
-                        if analysis_params[param] is not None
-                    }
-
-                    peaks, valleys = peak_detector(interpolated_well_data, **peak_detector_params)
-                    # needs to be converted to lists to be sent as json in response
-                    peaks_and_valleys[well] = [peaks.tolist(), valleys.tolist()]
-
-                elif len(pv_parquet_path) == 1:
-                    # need to remove nan values becuase peaks and valleys are different length lists
-                    peaks = peak_valleys_df[f"{well}__peaks"].dropna().tolist()
-                    valleys = peak_valleys_df[f"{well}__valleys"].dropna().tolist()
-                    # stored as floats in df so need to convert to int
-                    peaks_and_valleys[well] = [[int(x) for x in peaks], [int(x) for x in valleys]]
-                    logger.info(f"{len(peaks)} peaks and {len(valleys)} valleys for well {well}")
-
-                coordinates[well] = [
-                    [time[i] / MICRO_TO_BASE_CONVERSION, val] for i, val in enumerate(well_force)
-                ]
 
-            # Luci (12/14/2022) analysis_params["peaks_valleys"] will be a dictionary in version < 0.28.2 when peaks and valleys are only stored in this db column and not in s3
-            if analysis_params.get("peaks_valleys") is not None and isinstance(
-                analysis_params["peaks_valleys"], dict
-            ):
-                peaks_and_valleys = analysis_params["peaks_valleys"]
+            # Luci (03/21/2023) only get peaks and valleys on initial waveform request when interactive analysis is first opened
+            # this request will be called each time user switches between wells, we don't want to get peaks and valleys each time
+            peaks_and_valleys = None
+            if peaks_valleys_requested:
```

Comment:
It might make sense to break the peaks/valleys retrieval into its own route, but that's something we should probably all discuss together

Reply:
Yeah I thought about that too, just seemed like a bunch of duplicate code, but it would make it more readable and I like having routes with specific functionality.